core/src/Threads/Kokkos_ThreadsExec.cpp

```diff
@@ -399,27 +399,68 @@ bool ThreadsExec::wake() {
 
 //----------------------------------------------------------------------------
 
-void *ThreadsExec::root_reduce_scratch() {
-  return s_threads_process.reduce_memory();
-}
+void ThreadsExec::first_touch_allocate_thread_private_scratch_in_serial() {
```

Comment:
I think this one actually should be `resize_scratch_serial` (since it does the entire resize dealloc + alloc)